lib/queueClient.ts

```diff
@@ -0,0 +1,209 @@
+   * while receiving messages.
+   *
+   * @returns {ReceiveHandler} ReceiveHandler - An object that provides a mechanism to stop
+   * receiving more messages.
+   */
+  receive(onMessage: OnMessage, onError: OnError, options?: MessageHandlerOptions): ReceiveHandler {
+    if (!this._context.streamingReceiver ||
+      (this._context.streamingReceiver && !this._context.streamingReceiver.isOpen())) {
+      if (!options) options = {};
+      const rcvOptions: ReceiveOptions = {
+        maxConcurrentCalls: this.maxConcurrentCalls,
+        receiveMode: this.receiveMode,
+        autoComplete: options.autoComplete
+      };
+      const sReceiver = StreamingReceiver.create(this._context, rcvOptions);
+      this._context.streamingReceiver = sReceiver;
+      sReceiver.receive(onMessage, onError);
+      return new ReceiveHandler(sReceiver);
+    } else {
+      const rcvr = this._context.streamingReceiver;
+      const msg = `A "${rcvr.receiverType}" receiver with id "${rcvr.id}" has already been ` +
+        `created for the Queue "${this.name}". Another receive() call cannot be made while the ` +
+        `previous one is active. Please stop the previous receive() by calling ` +
+        `"receiveHandler.stop()".`;
+      throw new Error(msg);
+    }
+  }
+
+  /**
+   * Receives a batch of Message objects from a ServiceBus Queue for a given count and a
+   * given max wait time in seconds, whichever happens first.
+   * @param {number} maxMessageCount        The maximum message count. Must be a value greater than 0.
+   * @param {number} [maxWaitTimeInSeconds] The maximum wait time in seconds for which the Receiver
+   * should wait to receiver the said amount of messages. If not provided, it defaults to 60 seconds.
+   *
+   * @returns {Promise<Message[]>} A promise that resolves with an array of Message objects.
+   */
+  async receiveBatch(maxMessageCount: number, maxWaitTimeInSeconds?: number): Promise<Message[]> {
+    if (!this._context.batchingReceiver ||
+      (this._context.batchingReceiver && !this._context.batchingReceiver.isOpen()) ||
+      (this._context.batchingReceiver && !this._context.batchingReceiver.isReceivingMessages)) {
+      const options: ReceiveOptions = {
+        maxConcurrentCalls: 0,
+        receiveMode: this.receiveMode
+      };
+      const bReceiver: BatchingReceiver = BatchingReceiver.create(this._context, options);
+      this._context.batchingReceiver = bReceiver;
+      let error: MessagingError | undefined;
+      let result: Message[] = [];
+      try {
+        result = await bReceiver.receive(maxMessageCount, maxWaitTimeInSeconds);
+      } catch (err) {
+        error = err;
+        debug("[%s] Receiver '%s', an error occurred while receiving %d messages for %d max time:\n %O",
+          this._context.namespace.connectionId, bReceiver.id, maxMessageCount, maxWaitTimeInSeconds, err);
+      }
+      if (error) {
```

Comment:
why not just throw in the catch block rather than assigning to a variable and throwing immediately after?